### PR TITLE
New package: passwordsafe-0.97BETA

### PR DESCRIPTION
**Pull request by equalsraf**

### CODE REVIEW

**Comment by yopito** [comment]
I was also working on packaging PasswordSafe these days ! 
I have packaged some yubikey support. I'm waiting your pull request to be merged first before providing it.

File: srcpkgs/passwordsafe/template [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Template file for 'passwordwsafe'
+pkgname=passwordsafe
+version=0.9.7BETA
+revision=1
+wrksrc="pwsafe-0.97BETA"
```
Comment (xtraeme) [inline srcpkgs/passwordsafe/template]: version=0.9.7BETA vs wrksrc 0.97BETA

What's the correct one?

File: srcpkgs/passwordsafe/template [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+build_style=gnu-makefile
+# TODO: pkg dependencies to support yubikey
+make_build_args="NO_YUBI=1"
+make_build_target="release help"
+# Makefile uses perl to generate core_st.cpp, zip to generate help files
+hostmakedepends="perl wxWidgets-devel zip"
+makedepends="wxGTK-devel wxWidgets-devel libXt-devel libXtst-devel libxerces-c-devel libuuid-devel"
```
Comment (xtraeme) [inline srcpkgs/passwordsafe/template]: I really doubt this needs both wxWidgets versions, probably you can drop `wxGTK-devel`.

Comment (equalsraf) [inline srcpkgs/passwordsafe/template]: Removed.

File: srcpkgs/passwordsafe/template (hunk 1 above)
Comment (equalsraf) [inline srcpkgs/passwordsafe/template]: 0.97 is the correct one, moved version number into a variable.

**Comment by equalsraf** [comment]
@yopito go for it, I have a couple of rough templates for libyubikey and ykpers, but they are not ready yet. Also yubikey support is currently not working for me ([issue-1199](http://sourceforge.net/p/passwordsafe/bugs/1199/))

**Comment by yopito** [comment]
Sorry for a little off-topic post and late answer.

@equalsraf : Thanks for the pointers. I don't own yet a yubikey, so I can't check my yubikey support. Will you kindly test it once preliminary PR of passwordsafe will be done ? Or wait I buy one and understand how to use it ;)

My work (in progress) is available in branch `passwordsafe` of my fork https://github.com/yopito/void-packages.git, including some merge from your fork.

Please feel free to copy/paste what suits to you.
In particular, `srcpkgs/passwordsafe/*` contains patch and other stuff for xml resources you're currently missing.

As a reminder, here the list of files to check:
```
$ git branch 
passwordsafe
$ git diff master --name-only
common/shlibs
passwordsafe.pkg-notes.md
srcpkgs/libyubikey-devel
srcpkgs/libyubikey/template
srcpkgs/passwordsafe/patches/override-ldflags.patch
srcpkgs/passwordsafe/patches/passwordsafe-0.97-fix-resource-paths.patch
srcpkgs/passwordsafe/template
srcpkgs/ykpers-devel
srcpkgs/ykpers/template
```

**Comment by equalsraf** [comment]
Thanks  I missed the xml files and the locales. Not sure about the second patch though - the paths are a bit unusual but probably not worth patching.

I have not installed the locale files, they throw some [errors](https://gist.github.com/equalsraf/1905491fefba13af0a8e) when building.

xbps update-check was not working, since the passwordsafe git repo is now in github I just updated the distfiles, that ends up fixing update-check too.

PS: @yopito your ykpers packages did work, I was able to build passwordsafe and the tools are working, but they are missing the udev rules (needed to add permission to the usb device) e.g. https://github.com/equalsraf/void-packages/commit/50436558a6d78d094af3cbb84ce1c2c9b4f4cfe5#diff-faaf1a9aa6b285282ff09a7f438a0668R27

**Comment by yopito** [comment]
sorry for late reply.

> Not sure about the second patch though - the paths are a bit unusual but probably not worth patching.

to make the path of files more consistent: use `passwordsafe` everywhere, without a mix with `pwsafe`
Also avoid possible filename conflict with "pwsafe" project (http://nsd.dyndns.org/pwsafe).
I've updated my `passwordsafe/template` in this direction too (just commited)

> I have not installed the locale files, they throw some errors when building.

Yes; furthermore, locales are broken. I've noticed trange runtime behaviour when requesting change of GUI language (at least for French).
Need a deeper look, so excluding locales is a safe way for the moment ;)

> xbps update-check was not working, since the passwordsafe git repo is now in github I just updated the distfiles, that ends up fixing update-check too.

OK. my last commit merged as much as possible with your updates.
Please have a closer look and consider merging my proposals for `passwordsafe/template`:
* no need of `_version` I think
* build needs `pkg-config` and `gettext`: check from an empty masterdir to ensure this
* `patches/passwordsafe-0.97-fix-resource-paths.patch`
* project home page is http://pwsafe.org
* ...

> PS: @yopito your ykpers packages did work, I was able to build passwordsafe and the tools are working, but they are missing the udev rules (needed to add permission to the usb device) e.g. equalsraf@5043655#diff-faaf1a9aa6b285282ff09a7f438a0668R27

Great. Thanks for the pointer, merged it into my ykpers template.

**Comment by equalsraf** [comment]
Fair point about the paths, I have:removed _version, added the patch and dependencies, and updated the homepage.

File: srcpkgs/passwordsafe/template [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	vinstall install/graphics/pwsafe.png 644 usr/share/icons/hicolor/48x48/apps/
+	# Help files
+	for f in help/help*.zip; do
+		vinstall $f 644 usr/share/${pkgname}/help
+	done
+	# XML files
+	vmkdir usr/share/pwsafe
+	vcopy xml usr/share/${pkgname}
+	vlicense LICENSE
+	vdoc docs/ReleaseNotes.txt
```
Comment (xtraeme) [inline srcpkgs/passwordsafe/template]: Please do not install `ReleaseNotes.txt` nor `ChangeLog.txt`. Those are redundant.

Comment (equalsraf) [inline srcpkgs/passwordsafe/template]: Removed.

**Pull request merged**